### ISSUE TITLE
Question: Safest way to share Certificate between CNIs

### ISSUE THREAD

**Issue by RobOfTheSkyPeople** [boostchicken/udm-utilities#98]
Hi,
I love these resources. I was able to install several of them, but have run into a question I cannot yet solve: 

**What is the safest way to share the Let's Encrypt cert and key generated by the [UDM-LE](https://github.com/kchristensen/udm-le) container with the [AdGuardHome](https://github.com/boostchicken/udm-utilities/tree/master/AdguardHome) container?**

The UDM-LE script places a copy of the Let's Encrypt cert and key in the following directories:
	`/mnt/data/udm-le-master/lego/certificates/`
	`/mnt/data/unifi-os/unifi-core/config/`

But the AdGuardHome Container (via WebUI > Settings > Encryption Settings > Certificate Path) cannot seem to read either of those directories.

I'd like to know the most secure way to grant AdGuardHome access to the cert and key. Do I...
- Modify the UDM-LE script to also copy the new cert and key to some place AdGuardHome can access (not sure if AdguardHome-Confdir or AdguardHome-Workdir would be better)?
- Create some sort of symlink to one of the directories above?
- Modify how AdGuardHome runs/loads so it can mount one of those directories?

Specifically, I don't want to put the key in an unsafe location just to enable HTTPS for AdGuardHome.

**Comment by DaftHonk** [comment]
You want the third option. bind-mount the udm-le certicates folder into the adguardhome container. Add the following to your `podman run` command:
```
-v /mnt/data/udm-le-master/lego/certificates:/desired/cert/path
```
I usually just mount my certs to /ssl

**Comment by boostchicken** [comment]
@DaftHonk is correct.  Closing issue

**Comment by OverengineeredNetwork** [comment]
Thanks @boostchicken and @DaftHonk. I’ll give this a try soon. 

Before I do, I was curious if anyone knows what account is used to access the mounted directory. I am uneasy with the private key granting everyone read access. I would feel better if I could identify the account that needs access and adding it to a group so as to restrict access to the private key.

**Comment by OverengineeredNetwork** [comment]
Ok. I finally got around to looking at this again.

I started fresh by stopping and removing my existing AdGuardHome container
`podman stop adguardhome`
`podman rm adguardhome`

I pulled the latest AdGuardHome while I was at it
`podman pull adguard/adguardhome:latest`

I ran the install commands again with your recommendation
```
podman run -d --network dns --restart always  \
--name adguardhome \
-v "/mnt/data/AdGuardHome-Confdir/:/opt/adguardhome/conf/" \
-v "/mnt/data/AdGuardHome-Workdir/:/opt/adguardhome/work/" \
-v "/mnt/data/udm-le-master/lego/certificates/domain.tld/:/ssl/" \
--dns=127.0.0.1 --dns=1.1.1.1 \
--hostname adguardhome \
adguard/adguardhome:latest

```
I navigated to Settings > Encryption settings and saw that after checking Enable Encryption, the certificate and key path fields dynamically validate the certificate and key files. 
Of note, I had to use my fullchain.cer and not just the individual cert.

Once I got this working, I chmodded the cert and key files to 600 and restarted the adguardhome container and it still worked. So, I guess these containers run as root? Kinda scary.
